tests/requestparsing/contentlengthandtransferencoding_test.go

```diff
@@ -0,0 +1,347 @@
+				"Transfer-Encoding: identity\r\n" +
+				"\r\n" +
+				"5\r\n" +
+				"ABCDE\r\n" +
+				"0\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{"Content-Length": []string{"11"}},
+				contentLength:    11,
+				transferEncoding: nil,
+				body:             "5\r\nABCDE\r\n0",
+			},
+		},
+		{
+			name: "TransferEncodingListIdentityFirst",
+			request: []byte("GET / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Length: 11\r\n" +
+				"Transfer-Encoding: identity, xyz, asdf\r\n" +
+				"\r\n" +
+				"5\r\n" +
+				"ABCDE\r\n" +
+				"0\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{"Content-Length": []string{"11"}},
+				contentLength:    11,
+				transferEncoding: nil,
+				body:             "5\r\nABCDE\r\n0",
+			},
+		},
+		{
+			name: "TransferEncodingListChunkedIdentity",
+			request: []byte("GET / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Length: 11\r\n" +
+				"Transfer-Encoding: chunked, identity\r\n" +
+				"\r\n" +
+				"5\r\n" +
+				"ABCDE\r\n" +
+				"0\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{},
+				contentLength:    -1,
+				transferEncoding: []string{"chunked"},
+				body:             "ABCDE",
+			},
+		},
+		{
+			name: "TransferEncodingCasingOrdering1",
+			request: []byte("GET / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Transfer-Encoding: chunked\r\n" +
+				"transfer-Encoding: identity\r\n" +
+				"\r\n" +
+				"5\r\n" +
+				"ABCDE\r\n" +
+				"0\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{},
+				contentLength:    -1,
+				transferEncoding: []string{"chunked"},
+				body:             "ABCDE",
+			},
+		},
+		{
+			name: "TransferEncodingCasingOrdering2",
+			request: []byte("GET / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"transfer-Encoding: chunked\r\n" +
+				"Transfer-Encoding: identity\r\n" +
+				"\r\n" +
+				"5\r\n" +
+				"ABCDE\r\n" +
+				"0\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{},
+				contentLength:    -1,
+				transferEncoding: []string{"chunked"},
+				body:             "ABCDE",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), tt.request, func(r *http.Request) {
+				if diff := cmp.Diff(tt.want.headers, map[string][]string(r.Header)); diff != "" {
+					t.Errorf("r.Header mismatch (-want +got):\n%s", diff)
+				}
+
+				if diff := cmp.Diff(tt.want.transferEncoding, r.TransferEncoding); diff != "" {
+					t.Errorf("r.TransferEncoding mismatch (-want +got):\n%s", diff)
+				}
+
+				if r.ContentLength != tt.want.contentLength {
+					t.Errorf("r.ContentLength got: %v want: %v", r.ContentLength, tt.want.contentLength)
+				}
+
+				body, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					t.Errorf("ioutil.ReadAll(r.Body) got err: %v", err)
```

Comment:
I think this should be a `t.Fatal`. If you got an unespected I/O error it doesn't make much sense to assert anything on the returned body.